src/generic/TotalFraction.jl

```diff
@@ -391,18 +391,31 @@ end
 
 ###############################################################################
 #
-#   Unsafe operators and functions
+#   Unsafe functions
 #
 ###############################################################################
 
 function zero!(c::TotFrac)
    c.num = zero!(c.num)
    if !isone(c.den)
-      c.den = one(base_ring(c))
+      c.den = one!(c.den)
    end
    return c
 end
 
+function one!(c::TotFrac)
+   c.num = one!(c.num)
+   if !isone(c.den)
+      c.den = one!(c.den)
+   end
```

Comment:
I am wondering if it is cheaper to always set the denominator instead of doing this check?

Reply:
Probably depends. But fine by me to change it (I simply copied the code from `zero!`)